### DISCUSSION TITLE
Broken Truncate and/or Broken Anchor Underline styling

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#854] [Q&A] [answered]
Requested from: https://github.com/twilio-labs/paste/pull/818#issuecomment-708599779

See Sandbox for minimal reproduction: https://codesandbox.io/s/broken-truncateanchor-underline-styling-2b34z?file=/src/Form.tsx

Truncating an Anchor breaks the underline styling. [This is visible on the Paste website as well](https://paste.twilio.design/components/truncate/#truncate-url):
![image](https://user-images.githubusercontent.com/6936736/96043779-5570cf00-0e24-11eb-8cb3-8cc346403a0e.png)

As you can see in the Sandbox, I tried fixing this by adding some custom flex styling but this also removes the ellipsis from the Anchor.

**Comment by vankhoawin** [comment]
and it's not necessarily the underline that affects me, but without the underline the content appears to be misaligned (note the callback links on the right):
![image](https://user-images.githubusercontent.com/6936736/96044221-ee9fe580-0e24-11eb-838e-5dc422ec66f0.png)

**Comment by SiTaggart** [comment]
Thanks @vnguyen94, We'll investigate this further and get back to you.

**Comment by TheSisb** [comment, marked as answer]
It's not terribly clear and we should make the docs a bit better. Truncate adds a span to the DOM, and if you put a span inside an anchor element, the underline won't inherit. Therefore you should wrap the Anchor with the Truncate component.

Your third example looks fine when I remove the inline-style `{{display: "flex"}}`:
<img width="820" alt="image" src="https://user-images.githubusercontent.com/1592327/96166086-7f40f900-0eeb-11eb-81f9-d8698990e6e0.png">

Did I understand correctly?

**Reply by vankhoawin** [reply]
That did the trick, I swear I tried that combo 😅 Great looks like it works as expected, just that the example in doc is flipped as you mentioned. Thank you!
